File: frontend/src/constants/BookingStatus.ts
export const BOOKING_STATUSES = ["CONFIRMED", "PENDING", "RELEASED"] as const;
export type BookingStatusValue = (typeof BOOKING_STATUSES)[number];

export const BOOKING_STATUS_TEXT: Record<BookingStatusValue, string> = {
  CONFIRMED: "已确认",
  PENDING: "待处理",
  RELEASED: "已释放"
};

File: backend/src/constants/BookingStatus.go
package constants

// BookingStatus is the resource appointment state.
type BookingStatus string

const (
	BookingConfirmed BookingStatus = "CONFIRMED" // 已确认占用资源时段
	BookingPending   BookingStatus = "PENDING"   // 被挤出/待调度处理
	BookingReleased  BookingStatus = "RELEASED"  // 已释放
)

var BookingStatuses = []BookingStatus{
	BookingConfirmed, BookingPending, BookingReleased,
}

var BookingStatusText = map[BookingStatus]string{
	BookingConfirmed: "已确认",
	BookingPending:   "待处理",
	BookingReleased:  "已释放",
}

func (s BookingStatus) Valid() bool {
	for _, candidate := range BookingStatuses {
		if candidate == s {
			return true
		}
	}
	return false
}

func (s BookingStatus) Text() string { return BookingStatusText[s] }
